packages/hardhat/contracts/BuidlBadges.sol

```diff
@@ -59,24 +59,16 @@ contract BuidlBadges is ERC1155, Ownable, AccessControl {
         address recipient,
         uint256 tokenId
     ) public adminOnly {
-        _mint(msg.sender, tokenId, 1, "");
+        require((balanceOf(recipient, tokenId) == 0), "Already holds badge");
+        _mint(recipient, tokenId, 1, "");
 
         //After mint we set a bool for the tokenID to the user address.
         hasBadge[recipient][tokenId] = true;
```

Comment:
What is the idea behind this? I see that you are using `balanceOf` on the frontend, so not sure why we need this.

Reply:
This reverts the function if the user already holds that specific badgeId already.

It’s so we aren’t letting a user spam mint requests via SRE.

Reply:
Sorry, It seems I remarked the wrong lines. I meant the 

`mapping(address => mapping(uint => bool)) public hasBadge;`

I see that we are filling it on mint, but not using it on the front-end or anywhere else.